content/client-lib-development-guide/features.textile

```diff
@@ -1332,6 +1343,8 @@ h4. ChannelOptions
 ** @(TB2b)@ @cipher@, which is either:
 *** @(TB2b1)@ A @CipherParams@ instance, or
 *** @(TB2b2)@ an options hash (or language equivalent) consisting of any subset of @CipherParams@ fields that includes a @key@. In this case, the client library should call @getDefaultParams@, passing it the options hash, to obtain a @CipherParams@ instance
+** @(TB2c)@ @params@ (for realtime client libraries only) a @Dict<string,string>@ of key/value pairs
+** @(TB2d)@ @modes@ (for realtime client libraries only) a typed array of modes, which are the subset of flags defined in "@TR3@":#TR3 whose value is ≥16
```

Comment:
I am a bit unclear how `modes ` works (excuse my stupidity). Are we saying they are typed bit flags, and the end user (consumer of these APIs) needs to be able to do some basic bitwise & operations to work out which flags are set?

Reply:
> Are we saying they are typed bit flags, and the end user (consumer of these APIs) needs to be able to do some basic bitwise & operations to work out which flags are set?

No -- intention is for the client library to decode it into an array of elements of the ChannelMode enum. (Under the hood those'll be, shrug, whatever enum elements actually are the language in question). E.g. `[ChannelMode.Publish, ChannelMode.Subscribe]`.

The wording in TB2d isn't very clear, I'll clarify it. Clearer in the IDL (`modes?: [ChannelMode]`) and in RTL4m ("...the client library should decode the `flags` into an array of `ChannelMode`s...")